src/emit/instructions.rs

```diff
@@ -434,22 +436,14 @@ pub(super) fn process_instruction<'a, T: TargetRuntime<'a> + ?Sized>(
                 bin.context.i32_type().const_zero(),
             );
         }
-        Instr::AssertFailure { expr: Some(expr) } => {
-            let v = expression(target, bin, expr, &w.vars, function, ns);
-
-            let selector = 0x08c3_79a0u32;
-
-            let (data, len) = target.abi_encode(
-                bin,
-                Some(bin.context.i32_type().const_int(selector as u64, false)),
-                false,
-                function,
-                &[v],
-                &[Type::String],
-                ns,
-            );
+        Instr::AssertFailure {
+            encoded_args_with_len: Some(expr),
+        } => {
+            let data = expression(target, bin, &expr.0, &w.vars, function, ns);
+            let vector_bytes = bin.vector_bytes(data);
+            let len = expression(target, bin, &expr.1, &w.vars, function, ns);
```

Comment:
How about:
```rust
            let len = bin.vector_len(data);
```
And then we don't need the second expression in `Instr::AssertFailure { .. }`?

Reply:
That is also an option. I added both the buffer and the length in codegen, because the `encoder.abi_encode` function already returns the length of the buffer.